routes/index.js

```diff
@@ -0,0 +1,35 @@
+import express from 'express';
+import { fetchAllQuestionsCtrl } from '../controllers/fetchAllQuestions';
+import { fetchQuestionCtrl } from '../controllers/fetchQuestion';
+import { postQuestion } from '../controllers/postQuestion';
+import { postQuestionValidate } from '../middlewares/postQuestionValidate';
+import { postAnswer } from '../controllers/postAnswer';
+import { postAnswerValidate } from '../middlewares/postAnswerValidate';
+import { urlErrHandler } from '../middlewares/urlErrHandler';
+
+const router = express.Router();
+
+// GET ALL QUESTIONS
+router.get('/questions', fetchAllQuestionsCtrl);
+
+// GET A QUESTION
+router.get('/questions/:id([0-9]{1,})', fetchQuestionCtrl);
+
+// POST A QUESTION
+router.post('/questions', postQuestionValidate, postQuestion);
+
+// POST AN ANSWER
+router.post('/questions/:id([0-9]{1,})/answers', postAnswerValidate, postAnswer);
+
+
+// BAD GET A QUESTION REQUEST
+router.get('/questions/:id([A-z]{1,})', urlErrHandler);
```

Comment:
Validation and error handling is a very important part of any application and I can see you are trying to cater for that here but having a new route to handle that is not the best approach. You can make use of middleware on the routes above to check for bad requests and not have a separate bad request route for them. Let me know if you need more clarification on this.

Reply:
I will.
Thank you.